### PR TITLE
100 years twitter — review and manage command 

### PR DESCRIPTION
**Pull request by rlskoeser**
This is still a WIP but I finally came up with an approach for finding the events and displaying them on the review page that I'm happy with, and a solution that I think will work for scheduling and tweeting. I haven't written any unit tests and still have a few more refinements to do, but thought maybe you could review at your convenience to see if you have any concerns so I can make myself a checklist and merge when I finish adding tests etc.

### CODE REVIEW

File: mep/accounts/management/commands/twitterbot_100years.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        # auth.set_access_token('', '')
+        # api = tweepy.API(auth)
+        # api.update_status(content)
+
+
+tweetable_event_types = ['Subscription', 'Renewal', 'Reimbursement',
+                         'Borrow', 'Purchase', 'Request']
+
+
+def can_tweet(ev, date):
```
Comment (thatbudakguy) [inline mep/accounts/management/commands/twitterbot_100years.py:137]: could these be `@classmethod`s of the `Command`? idk if it makes much of a difference, they just still feel associated to me

File: mep/accounts/management/commands/twitterbot_100years.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        print(content)
+        # TODO: set credentials in local settings
+        # auth = tweepy.OAuthHandler('', '')
+        # auth.set_access_token('', '')
+        # api = tweepy.API(auth)
+        # api.update_status(content)
+
+
+tweetable_event_types = ['Subscription', 'Renewal', 'Reimbursement',
+                         'Borrow', 'Purchase', 'Request']
```
Comment (thatbudakguy) [inline mep/accounts/management/commands/twitterbot_100years.py:134]: whoa, what's a Request?

File: mep/accounts/management/commands/twitterbot_100years.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+       ev.subscription.purchase_date == event_date:
+        content = '%s joined the Shakespeare and Company lending library.' % \
+            member_name
+
+    elif event_label in ['Borrow', 'Purchase', 'Request']:
+        work = work_label(ev.work)
+        # convert event type into verb for the sentence
+        verb = '%sed' % ev.event_type.lower().rstrip('e')
+        if event_label == 'Borrow' and ev.end_date == event_date:
+            verb = 'returned'
```
Comment (thatbudakguy) [inline mep/accounts/management/commands/twitterbot_100years.py:188]: why do we do returns but not borrows?

File: mep/accounts/management/commands/twitterbot_100years.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        content = '%s renewed for %s' % \
+            (member_name, ev.subscription.readable_duration())
+        # include volume count if known
+        if ev.subscription.volumes:
+            content = '%s at %d volume%s per month.' % \
+                (content, ev.subscription.volumes,
+                 '' if ev.subscription.volumes == 1 else 's')
+
+    elif event_label == 'Reimbursement':
+        # renewed for 2 months at 1 volume per month
```
Comment (thatbudakguy) [inline mep/accounts/management/commands/twitterbot_100years.py:203]: think this comment was copy-pasted from above but not edited to match the reimbursement message format

File: mep/accounts/management/commands/twitterbot_100years.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        content = '%s received a reimbursement for %s%s from the Shakespeare and Company lending library' % \
+            (member_name, ev.reimbursement.refund,
+             ev.reimbursement.currency_symbol())
+
+    # any other kind of event: no tweet
+
+    if content:
+        # change: use card detail url if possible
```
Comment (thatbudakguy) [inline mep/accounts/management/commands/twitterbot_100years.py:211]: agree, card detail url would be awesome!

Comment (thatbudakguy) [reply mep/accounts/management/commands/twitterbot_100years.py:211]: oh, it looks like you already do. great!

File: requirements.txt [hunk 6]
```diff
@@ -25,4 +25,5 @@ bleach
 django-fullurl
 unidecode
 stop_words
-django-markdownify
\ No newline at end of file
+django-markdownify
```
Comment (thatbudakguy) [inline requirements.txt:28]: what's this used for?

File: mep/accounts/management/commands/twitterbot_100years.py (hunk 2 above)
Comment (rlskoeser) [reply mep/accounts/management/commands/twitterbot_100years.py:134]: It's one of the new custom types that can be set based on notation in the private notes!

File: mep/accounts/management/commands/twitterbot_100years.py (hunk 1 above)
Comment (rlskoeser) [reply mep/accounts/management/commands/twitterbot_100years.py:137]: Hmm, will consider. Agree they are associated. I wanted to make the `tweet_content` method available for use in the template tag, but a class method wouldn't preclude that.

File: mep/accounts/management/commands/twitterbot_100years.py (hunk 3 above)
Comment (rlskoeser) [reply mep/accounts/management/commands/twitterbot_100years.py:188]: a regular borrow is handled by the default logic (event => verb with start date)

File: mep/accounts/management/commands/twitterbot_100years.py (hunk 4 above)
Comment (rlskoeser) [reply mep/accounts/management/commands/twitterbot_100years.py:203]: ha, thanks, yes

File: requirements.txt (hunk 6 above)
Comment (rlskoeser) [reply requirements.txt:28]: markdownify ? was an easy way to allow for simple formatting in public notes

**Comment by rlskoeser** [comment]
Cleaned up the todos and fixmes, added unit tests for everything and corrected or refined a few more things along the way.

**Pull request merged**